Formula/cacert.rb

```diff
@@ -0,0 +1,87 @@
+class Cacert < Formula
+  desc "Mozilla CA certificate store"
+  homepage "https://curl.se/docs/caextract.html"
+  url "https://curl.se/ca/cacert-2021-07-05.pem"
+  # Ideally we'd have a HTTP mirror, but a GitHub-hosted one is second best.
+  mirror "https://gist.githubusercontent.com/jonchang/b13c60cd50bdd91fd12a034ea53e39d5/raw/e1c430fd9ab59ed34212083537be0da974c26555/cacert-2021-07-05.pem"
+  sha256 "a3b534269c6974631db35f952e8d7c7dbf3d81ab329a232df575c2661de1214a"
+  license "MPL-2.0"
+
+  def install
+    pkgshare.install "cacert-#{version}.pem" => "cacert.pem"
+  end
+
+  def post_install
+    # macOS prior to 10.12.1 doesn't have the Let's Encrypt root, which is commonly used.
+    if OS.mac? && MacOS.version >= :sierra
+      post_install_from_keychain
+    else
+      post_install_from_pem
+    end
+  end
+
+  def post_install_from_keychain
+    ohai "Regenerating CA certificate bundle from keychain, this may take a while..."
+
+    keychains = %w[
+      /Library/Keychains/System.keychain
+      /System/Library/Keychains/SystemRootCertificates.keychain
+    ]
+
+    certs_list = `/usr/bin/security find-certificate -a -p #{keychains.join(" ")}`
+    certs = certs_list.scan(
+      /-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----/m,
+    )
+
+    # Check that the certificate has not expired
+    valid_certs = certs.select do |cert|
+      IO.popen("/usr/bin/openssl x509 -inform pem -checkend 0 -noout &>/dev/null", "w") do |openssl_io|
+        openssl_io.write(cert)
+        openssl_io.close_write
+      end
+
+      next unless $CHILD_STATUS.success?
+
+      # XXX And drop Kerberos certs which may invalidate the whole trust store
+      # due to bug in gnutls (https://gitlab.com/gnutls/gnutls/-/issues/1255)
+      IO.popen("/usr/bin/openssl x509 -inform pem -issuer -noout 2>/dev/null", "r+") do |openssl_io|
+        openssl_io.write(cert)
+        openssl_io.close_write
+        cn = openssl_io.read
+        openssl_io.close_read
+        cn.exclude? "CN=com.apple.kerberos.kdc"
+      end
+    end
+
+    # Check that the certificate is trusted in keychain
+    trusted_certs = begin
+      tmpfile = Tempfile.new
+
+      valid_certs.select do |cert|
+        tmpfile.rewind
+        tmpfile.write cert
+        tmpfile.truncate cert.size
+        tmpfile.flush
+        IO.popen("/usr/bin/security verify-cert -l -L -R offline -c #{tmpfile.path} &>/dev/null")
+
+        $CHILD_STATUS.success?
+      end
+    ensure
+      tmpfile&.close!
+    end
+
+    pkgetc.mkpath
+    (pkgetc/"cert.pem").atomic_write(trusted_certs.join("\n") << "\n")
```

Comment:
Is there any reason not to add the curl.se `cert.pem` here? May as well make use of the extra stuff you now download with this formula.

Reply:
Hmm maybe. I'll compare them and see what the current differences are.

Reply:
I'll get this to you tomorrow. Just realised it's 4.30am 😅

Reply:
https://gist.github.com/Bo98/3717e3384442e454b38be0259d094e4d
https://gist.github.com/Bo98/e958f5b3c2a23ed338de94e0a3b6e248

Reply:
(Note that some differences will be because Apple are slower to remove expired certs)

Reply:
Added